phase/exporter.go

```diff
@@ -281,54 +279,51 @@ func (e *Exporter) addExtensionLayers(opts ExportOptions) error {
 		return err
 	}
 	history := configFile.History
-	var (
-		localImage   bool
-		artifactsDir string
-	)
-	if isLocalImage(opts.WorkingImage) {
-		localImage = true
-		if artifactsDir, err = os.MkdirTemp("", "lifecycle.exporter.layer"); err != nil {
-			return err
-		}
-	}
-	for idx, l := range extendedLayers {
-		layerHex, err := l.DiffID()
+	for idx, l := range allLayers {
+		layerDigest, blobExists, err := getLayerInfo(l, extendedRunImagePath)
 		if err != nil {
-			if _, ok := err.(*fs.PathError); ok {
-				continue // failed to get the diffID because the blob doesn't exist
-			}
-			return err
-		}
-		digest, err := l.Digest()
-		if err != nil {
-			return err
+			return fmt.Errorf("failed to get layer info: %w", err)
 		}
-		layerPath := filepath.Join(extendedRunImagePath, "blobs", digest.Algorithm, digest.Hex)
-		if localImage {
-			var calculatedDiffID string
-			if calculatedDiffID, err = uncompressLayerAt(layerPath, artifactsDir); err != nil {
-				return err
-			}
-			if calculatedDiffID != layerHex.String() {
-				return fmt.Errorf("digest of uncompressed layer from %s does not match expected value; found %q, expected %q", layerPath, calculatedDiffID, layerHex.String())
-			}
-			layerPath = filepath.Join(artifactsDir, calculatedDiffID)
+		if !blobExists {
```

Comment:
If the blob doesn't exist we are skipping it and not throwing an error - is that expected? If so, maybe a comment explaining why would be good for future us!

Reply:
Added a comment in https://github.com/buildpacks/lifecycle/pull/1289/commits/51050acd241dc0150df004bf17f08218b0ecb2a3